examples/imgui_impl_x11.cpp

```diff
@@ -0,0 +1,346 @@
+// dear imgui: Platform Binding for X11 using xcb
+// This needs to be used along with a Renderer (e.g. OpenGL3, Vulkan..)
+
+// This works as is with Vulkan. For OpenGL using GLX, you need a hybrid
+// of XLib and xcb using the X11/Xlib-xcb.h header. Use XLib to create the
+// GLX context, then use functions in Xlib-xcb.h to convert the XLib
+// structures to xcb, which you can then pass unmodified here.
+// Requires libxcb, libxcb-xfixes, libxcb-xkb1 -libxcb-cursor0 and libxcb-keysyms1
+
+// Implemented features:
+//  [X] Platform: Keyboard arrays indexed using XK symbols, e.g. ImGui::IsKeyPressed(XK_space).
+//  [X] Platform: Mouse cursor shape and visibility.
+// Missing features:
+//  [ ] Platform: Clipboard support
+//  [ ] Platform: Gamepad support. Enabled with 'io.ConfigFlags |= ImGuiConfigFlags_NavEnableGamepad'.
+
+#include "imgui.h"
+#include <X11/keysym.h>
+#include <xcb/xcb.h>
+#include <xcb/xcb_keysyms.h>
+#include <xcb/xfixes.h>
+#include <xcb/xkb.h>
+#include <xcb/xcb_cursor.h>
+#include <time.h>
+
+// CHANGELOG
+// (minor and older changes stripped away, please see git history for details)
+// 2020-07-30 Initial implementation
+
+// X11 Data
+static xcb_connection_t*     g_Connection;
+static xcb_drawable_t*       g_Window;
+static xcb_key_symbols_t*    g_KeySyms;
+static xcb_cursor_context_t* g_CursorContext;
+static xcb_screen_t*         g_Screen;
+
+static timespec             g_LastTime;
+static timespec             g_CurrentTime;
+
+static bool                 g_HideXCursor = false;
+static ImGuiMouseCursor     g_CurrentCursor = ImGuiMouseCursor_Arrow;
+
+static const char* g_CursorMap[ImGuiMouseCursor_COUNT] = {
+    "arrow",               // ImGuiMouseCursor_Arrow
+    "xterm",               // ImGuiMouseCursor_TextInput
+    "fleur",               // ImGuiMouseCursor_ResizeAll
+    "sb_v_double_arrow",   // ImGuiMouseCursor_ResizeNS
+    "sb_h_double_arrow",   // ImGuiMouseCursor_ResizeEW
+    "bottom_left_corner",  // ImGuiMouseCursor_ResizeNESW
```

Comment:
These are the defined X11 [standard names](https://tronche.com/gui/x/xlib/appendix/b/) for the different cursors. In true Linux tradition some don't accommodate all of them. For instance with KDE, and only KDE, the name of the corner resize cursors is `size_bdiag` and `size_fdiag` and KDE doesn't map those back to the standard `bottom_left_corner` and `bottom_right_corner`. It's not worth our time figuring all that out, just use the standard X names.